### PR TITLE
External provider query error

### LINKED ISSUE #1678: Missing information about failed query against external providers
**Describe the bug**
While working at rest issue  [#8327](https://github.com/DSpace/DSpace/issues/8327) we noticed that, in case of error during external system query, user is not properly informed about this failure.

**To Reproduce**
1. login to dspace7 and go to MyDSpace
2. start a new Publication submission from external source
3. enter a query string and select “Pubmed” as external source
4. In case query fails with a 500 backend error, following page is displayed

image
image

**Expected behavior**
In case of rest error (500 in this example) A message indicating that search went in error.  We believe current behavior could be improved, and such improvement would lead to a little UX improvement in case of failure of a query for live import towards an external system.

We, 4Science, can take in charge this improvement, our work estimation is 2/4 hours.

### PR DESCRIPTION
**Pull request by corrad82-4s**
## References
_Add references/links to any related issues or PRs. These may include:_
* Fixes #1678

## Description
Short summary of changes (1-2 sentences).
a new `<div>` has been added to html component for import from external sources, displaying an ad-hoc error message whenever REST call to query external system returns a 500 http error code.

List of changes in this PR:
* added a `div` providing an error message in case of 500 http error code returned by REST endpoint performing live import query
* added a test to check correct display of above mentioned `div` in case of 500 error.

**Include guidance for how to test or review your PR.** This may include: steps to reproduce a bug, screenshots or description of a new feature, or reasons behind specific changes. 
The 500 error code behavior is hard to reproduce, depending on possible error on third party systems. A proper test is provided to check correct behavior when REST call performing the live import query returns such an error.

## Checklist

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [X] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [X] My PR doesn't introduce circular dependencies
- [ ] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [X] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

File: src/app/submission/import-external/submission-import-external.component.spec.ts [hunk 1]
```diff
@@ -177,6 +184,326 @@ describe('SubmissionImportExternalComponent test suite', () => {
     });
   });
 
+  describe('handle BE response for search query', () => {
```
Comment (tdonohue) [inline src/app/submission/import-external/submission-import-external.component.spec.ts:187]: What does "BE" stand for here?  (After more thought, I realize you probably mean "backend")

Could we change this text to just say "Handle response for search query"?  Or "handle backend response..."?  I think abbreviations should be avoided unless they are very obvious (and if I was confused by "BE" then others likely would be as well)

File: src/app/submission/import-external/submission-import-external.component.spec.ts [hunk 2]
```diff
@@ -177,6 +184,326 @@ describe('SubmissionImportExternalComponent test suite', () => {
...
+        },
+        'pageInfo': {
+          'elementsPerPage': 10,
+          'totalElements': 11971608,
+          'totalPages': 1197161,
+          'currentPage': 1
+        },
+        '_links': {
+          'first': {
+            'href': 'https://dspacecris7.4science.cloud/server/api/integration/externalsources/scopus/entries?query=test&page=0&size=10&sort=id,asc'
```
Comment (tdonohue) [inline src/app/submission/import-external/submission-import-external.component.spec.ts:206]: Could we avoid using 4Science specific URLs in these tests?  If you want you can use https://demo7.dspace.org.  Or you could just use a fake URL like "https://rest.api/server/" (which some other tests use).

This same comment goes for all the "dspacecris7.4science.cloud" URLs in this PR.

**Review by tdonohue — COMMENTED** [review]
@corrad82-4s : Thanks!  This looks good overall.  I also tested it & it works...by chance I was able to reproduce a 500 error in PubMed just by running many searches in a row.

I'm basically +1, but I'd appreciate it if we could do some minor cleanup of the tests in this PR before merging. See my inline comments below.

File: src/app/submission/import-external/submission-import-external.component.spec.ts (hunk 2 above)
Comment (mwoodiupui) [reply src/app/submission/import-external/submission-import-external.component.spec.ts:206]: I'll just mention that `example.com` is permanently reserved for just such uses.

File: src/app/submission/import-external/submission-import-external.component.spec.ts (hunk 1 above)
Comment (corrad82-4s) [reply src/app/submission/import-external/submission-import-external.component.spec.ts]: Hi @tdonohue , description has been updated

File: src/app/submission/import-external/submission-import-external.component.spec.ts (hunk 2 above)
Comment (corrad82-4s) [reply src/app/submission/import-external/submission-import-external.component.spec.ts]: Hi @tdonohue , @mwoodiupui , 
thank you for pointing this out. I have just pushed a fix using example.com as domain. Thank you

**Review by tdonohue — APPROVED** [review]
👍 Thanks @corrad82-4s !  This all looks good to me now.

**Comment by tdonohue** [comment]
Merging as this was previously flagged as `1 Approval` and is at +1

**Pull request merged**